.circleci/config.yml

```diff
@@ -327,9 +334,9 @@ workflows:
   silkworm:
     jobs:
       - linux-gcc-12-release
-      - linux-gcc-11-conan
+      - linux-gcc-13-conan
       - linux-gcc-11-thread-sanitizer
-      - linux-clang-13-coverage
```

Comment:
I'd rather not remove clang 13, since otherwise we might break compilation with it in the future. To my mind we can use this PR as an opportunity to improve our CI a little bit. I suggest to have jobs like clang-min and clang-latest, similar to evmone. Our README says that building silkworm requires "GCC >= 11.2.0 or Clang >= 12.0.0". We should check those versions in CI or update README if we in fact require higher versions.

Reply:
The idea is to shift supported compiler versions from clang 13-15 to 14-16, and gcc 11-12 to 12-13.
So in addition to supporting the new compilers (clang 16 and gcc 13) this PR drops support of old clang 13 and gcc 11.

The reason I'm keeping linux-gcc-11-thread-sanitizer for now is Hunter and its old abseil. This job can be bumped to gcc 13 when we move it to conan, and the first step in that direction is this PR: https://github.com/torquem-ch/silkworm/pull/1078

As for clang, "clang-min" becomes 14, and I agree that ideally we should have a job that builds with clang 14. The reason I didn't add it is that all our core developers are using Xcode toolchain currently based on clang 14. So the benefit of having it in CI is fairly low, while the benefit of not having an extra job seems tangible.

I understand that your idea is to have a clang 14 job instead of clang 15, because if we build fine with v14 and v16 the chance that we break on v15 is low. I'm fine with that, although it would mean that in practice we would never build on clang 15, so we don't really know if it works (until Xcode upgrades to it), and at the same time we would build on clang 14 even more than now, which feels redundant.

Thanks for reminding about the readme, I've updated it accordingly now.

Reply:
Also note that having a compiler version in the job name separates the build caches at the moment. We can definitely rename jobs to something like clang-min and clang-latest, but need to remember to revise the cache keys.

Reply:
But why drop support of clang 12/13 and gcc 11 unless we want to use some C++ features not supported by the old compilers? The wider range of compilers we support, the better. For example, gcc11 is the default on the latest Ubuntu LTS.

Reply:
Good point. Do we aim to support building on Ubuntu LTS with a default compiler?

Reply:
@yperbasis I've reverted this change until further discussion in order to unblock this PR. I'm not against your proposal, but I'd like to have everyone on the same page before proceeding.

Reply:
OK we can merge it as it is now and discuss our compiler compatibility strategy separately